src/Couchbase.Lite.Shared/API/Database/DatabaseConfiguration.cs

```diff
@@ -107,7 +124,11 @@ public EncryptionKey? EncryptionKey
         /// </summary>
         public DatabaseConfiguration()
         {
-
+#if !MACCATALYST
+            if(!RuntimeInformation.IsOSPlatform(OSPlatform.OSX)) {
+                _mmapEnabled = Constants.DefaultDatabaseMmapEnabled;
+            }
+#endif
```

Comment:
From the CBL Discussion meeting, we don't need to implementation any guardrails or special logic for macOS. LiteCore will decide what to do. I am wondering why not including MACCATYLYST?

Reply:
After all, if you feel strongly that you want to do this, I don't have the problem at all.